domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgress.kt

```diff
@@ -14,8 +16,15 @@ private const val TERMINAL_INTERACTION_ID = "EndExploration"
  * instances, but calling code is responsible for ensuring it is properly reset.
  */
 internal class ExplorationProgress {
+  internal lateinit var currentProfileId: ProfileId
+  internal lateinit var currentTopicId: String
+  internal lateinit var currentStoryId: String
```

Comment:
Also, do we really need to add 3 different variables? Seems like a lot of added dependency.

Reply:
explorationId was already preset for loading the exploration, profiled, topicId and storyId are added to mark the chapter as IN_PROGRESS_SAVED/NOT_SAVED. ProfileId is also used to save checkpoints.